util/raclgen/lib.py

```diff
@@ -50,3 +57,46 @@ def compute_policy_value(permission: str) -> int:
             policy['rd_default'] = compute_policy_value('allowed_rd')
             policy['wr_default'] = compute_policy_value('allowed_wr')
     return racl_config
+
+
+def parse_racl_mapping(racl_config: Dict[str, object], mapping_path: str,
+                       if_name: Optional[str], ip_block: IpBlock):
+    mapping = _read_hjson(mapping_path)
+    parsed_register_mapping = {}
+
+    # Mapping must be a dict with a single entry:
+    # RACL_GROUP => register mapping
+    if len(mapping) != 1:
+        raise SystemExit('Mapping file must be a single-element dict mapping the RACL group to the '
+                         'register mapping')
+    racl_group = list(mapping.keys())[0]
+    register_mapping = list(mapping.values())[0]
+
+    # Special handling of the all star assignment:
+    # "*": POLICY
+    # Assigns all registers to a given policy
+    if len(register_mapping) == 1 and list(register_mapping.keys())[0] == "*":
+        policy_name = list(register_mapping.values())[0]
+
+        if racl_group not in racl_config['policies']:
+            raise SystemExit(f'RACL group {racl_group} not defined in RACL config')
+
+        policy_idx = -1
+        for idx, policy in enumerate(racl_config['policies'][racl_group]):
+            if policy['name'] == policy_name:
+                policy_idx = idx
+                break
+
+        if policy_idx == -1:
+            raise SystemExit(f'RACL policy {policy_name} not defined in RACL config')
+
+        if if_name not in ip_block.reg_blocks:
```

Comment:
I'd probably be inclined to assign `ip_block.reg_blocks.get(if_name)` to something and then check for `None` here, which avoids needing to do the look-up again as the next operation.